### PR TITLE
feat(suite): Add anchor to Button if it's needed

### PR DESCRIPTION
**Pull request by jvaclavik**
## Description

This allows us to prevent components where `button` is inside of `a` like this one:
image

## Related Issue

Resolve

### CODE REVIEW

File: packages/components/src/components/buttons/Button/Button.tsx [hunk 1]
```diff
@@ -26,6 +27,7 @@ type ButtonContainerProps = TransientProps<AllowedFrameProps> & {
     $hasIcon?: boolean;
     $isFullWidth?: boolean;
     $isSubtle: boolean;
+    as?: 'a' | 'button';
```
Comment (seibei-iguchi) [inline packages/components/src/components/buttons/Button/Button.tsx:29]: What's the reason for this? Isn't it already decided by the presence of a href prop?

Comment (jvaclavik) [reply packages/components/src/components/buttons/Button/Button.tsx:29]: We need to change tag to `a` when we want to use `href`. it's not working on `button` AFAIK

Comment (seibei-iguchi) [reply packages/components/src/components/buttons/Button/Button.tsx:29]: Yeah but I don't see you use it anywhere? Shouldn't the styled ButtonContainer change to something else than button?

**Review by seibei-iguchi — APPROVED** [review]

File: packages/components/src/components/buttons/Button/Button.tsx (hunk 1 above)
Comment (jvaclavik) [reply packages/components/src/components/buttons/Button/Button.tsx:29]: Solved in DM. it's native feature of styled components

**Pull request merged**